app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,69 @@
+class OrdersController < ApplicationController
+  def index
+    @orders = Order.for_user(current_user)
+  end
+
+  def show
+    @order = Order.find(params[:id])
+  end
+
+  def no_cart
+    flash[:alert] = 'you have not put anything in the cart'
+    redirect_to root_path
+  end
+
+  def create
+    if current_user.present?
+      order = find_order
+      if order.blank?
+        flash[:alert] = "This Order is not belongs to you"
+        redirect_to :back
+        return
+      end
+
+      quantity = params[:order_detail][:quantity]
+      if quantity !=~ /^\d+$/ && quantity.to_i < 0
```

Comment:
Don't use '!' it will make it evaluates to true no matter what. Either

```ruby
!!(quantity !=~ /^\d+$/)
```

Which is cryptic, or:

```ruby
(quantity !=~ /^\d+$/).nil?
```